airflow/jobs.py

```diff
@@ -2540,9 +2542,25 @@ def _execute(self, session=None):
                         self.dag_id
                     )
                     time.sleep(self.delay_on_limit_secs)
+
+        except (KeyboardInterrupt, SystemExit):
+            for dag_run in self.dag_runs:
+                dag_run.refresh_from_db(session)
+                make_transient(dag_run)
+                unfinished_tasks = dag_run.get_task_instances(
+                    state=State.unfinished(),
+                    session=session
+                )
+                if unfinished_tasks:
+                    # if there are unfinished tasks and ctrl+9
+                    # set dag run state to failed
+                    dag_run.state = State.FAILED
+                else:
+                    dag_run.state = State.SUCCESS
```

Comment:
What if the DAG has failed tasks? We wouldn't want to mark the DagRun as SUCCESS in that case, right?

Reply:
you are right...I thought the unfinished state should include failed task as well. let me do an update.

Reply:
How about we set the `DagRun` to `State.FAILED` for all `DagRun` that is not in `State.SUCCESS` state?